src/dolphin/card/CARDCreate.c

```diff
@@ -112,9 +122,12 @@ s32 CARDCreateAsync(s32 chan, const char* fileName, u32 size,
     fileInfo->chan = chan;
     fileInfo->fileNo = freeNo;
 
-    result = __CARDAllocBlock(chan, size / card->sectorSize, CreateCallbackFat);
-    if (result < 0) {
+    /// @todo Eliminate cast to #CARDCallback.
+    result = __CARDAllocBlock(chan, size / card->sectorSize,
+                              (CARDCallback) CreateCallbackFat);
```

Comment:
Seems like you can remove these CARDCallback casts if you keep propagating EXIChannel to these function definitions

Reply:
Yeah I know, just seemed out of scope for this PR.

Reply:
Ok that's fair